src/wp-includes/capabilities.php

```diff
@@ -1062,6 +1074,99 @@ function wp_maybe_grant_site_health_caps( $allcaps, $caps, $args, $user ) {
 	return $allcaps;
 }
 
+function wp_site_health_allow_single_health_check( $user_id, $test_slug ) {
+	if ( ! is_multisite() ) {
+		$can_view = user_can( $user_id, 'view_site_health_checks' );
```

Comment:
:1234: This isn't needed as if you define it as a required cap in `map_meta_caps`, that will save having to go through the entire `user_can()` call again.